forge-game/src/main/java/forge/game/card/CardProperty.java

```diff
@@ -1188,7 +1189,18 @@ public static boolean cardHasProperty(Card card, String property, Player sourceC
         } else if (property.startsWith("dealtDamagetoAny")) {
             return card.getDamageHistory().getHasdealtDamagetoAny();
         } else if (property.startsWith("attackedThisTurn")) {
-            if (!card.getDamageHistory().getCreatureAttackedThisTurn()) {
+            if (card.getDamageHistory().getCreatureAttacksThisTurn() == 0) {
+                return false;
+            }
+        } else if (property.startsWith("attackedYouThisTurn")) {
+            boolean found = false;
+            for (PlayerCollection players : game.getPlayersAttackedThisTurn().values()) {
```

Comment:
right, got confused by the other PR :/

so I'll probably have to convert `Player.creatureAttackedThisTurn` into a `List<Pair<Card, GameEntity>>` then?

Should I try to make it cleaner with `Equivalence.Wrapper` too?

Reply:
we first need to decide if we want the Attacked This Turn be stored in the Player or Game Object.

because if we store the List<GameEntity> in Card(History) then we could better filter if Card Attacked Player

Reply:
Hmn, might be problematic for _Robber of the Rich_ 

would have to check all game zone cards if they contain LKI with Rogue and keep it copied with each zone change?

Reply:
`Jabari's Influence` needs to look on the Creature if it attacked a GameEntity (You)

`Robber of the Rich` would be a different List, it needs to look at You, the Player, and what creatures you attacked with (LKI)

Reply:
thanks, got it I think

that also helped to clean up _Port Razer_ nicely